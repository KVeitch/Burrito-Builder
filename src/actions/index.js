export const setOrders = orders => ({
  type: 'SET_ORDERS',
  orders
});

export const addOrder = order => {
  return ({
    type: 'ADD_ORDER',
    order
  })
}