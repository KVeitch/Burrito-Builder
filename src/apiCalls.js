export const getOrders = () => {
  return fetch('http://localhost:3001/api/v1/orders')
      .then(response => response.json())
}



export const postOrder = async newOrder => {
  const url = 'http://localhost:3001/api/v1/orders';
  const options = {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json'
    },
    body: JSON.stringify({ ...newOrder })
  };
  const response = await fetch(url, options);
  if (!response.ok) {
    throw new Error('Sorry.  Unable to create your new order.');
  }
  const returnedOrder = await response.json();
  return returnedOrder;
}