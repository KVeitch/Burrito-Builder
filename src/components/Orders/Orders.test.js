import React from 'react';
import { shallow } from 'enzyme';
import { Orders, mapDispatchToProps, mapStateToProps } from './Orders';
import * as actions from '../../actions'
import { getOrders } from '../../apiCalls';


jest.mock('../../apiCalls');

const mockOrders = [
  {
    id: 1,
    name: 'Pat',
    ingredients: [
      'beans',
      'lettuce',
      'carnitas',
      'queso fresco',
      'jalapeno'
    ]
  },
  {
    id: 2,
    name: 'Sam',
    ingredients: [
      'steak',
      'pico de gallo',
      'lettuce',
      'carnitas',
      'queso fresco',
      'jalapeno'
    ]
  },
  {
    id: 3,
    name: 'Alex',
    ingredients: [
      'sofritas',
      'beans',
      'sour cream',
      'carnitas',
      'queso fresco'
    ]
  }
]
getOrders.mockImplementation(() => Promise.resolve(mockOrders))

const mockMath = Object.create(global.Math);
mockMath.random = () => 0.5;
global.Math = mockMath;

describe('Orders', () => {
  let wrapper;
  beforeEach(()=>{
    wrapper = shallow(<Orders orders={mockOrders} setOrders={actions.setOrders}/>)
  })

  it('should match the snapshot',() => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should run getOrders after mounting', ()=>{
    expect(getOrders).toHaveBeenCalled()
  })



})

describe('Orders mapDispatchToProps', () => {
  it('should call dispatch with setOrder action when setOrders is called', () => {
       const mockDispatch = jest.fn();

       const actionToDispatch = actions.setOrders('SET_ORDERS',mockOrders);
       const mappedProps = mapDispatchToProps(mockDispatch);
       mappedProps.setOrders('SET_ORDERS',mockOrders);
       expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
       });
    });

  describe('Orders mapStateToProps', () => {
    it('mapStateToProps give all the needed data from state', () => {
        const mockState = {
          orders:mockOrders,
          odd:'NOthing else goes in state'
        };
  
        const expected = {
            orders:mockOrders
        };

        const mappedProps = mapStateToProps(mockState);
        expect(mappedProps).toEqual(expected);
    });
});