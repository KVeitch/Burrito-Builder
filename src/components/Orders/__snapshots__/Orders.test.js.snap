// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orders should match the snapshot 1`] = `
<section>
  <div
    className="order"
    key="0.5"
  >
    <h3>
      Pat
    </h3>
    <ul
      className="ingredient-list"
    >
      <li
        key="0.5"
      >
        beans
      </li>
      <li
        key="0.5"
      >
        lettuce
      </li>
      <li
        key="0.5"
      >
        carnitas
      </li>
      <li
        key="0.5"
      >
        queso fresco
      </li>
      <li
        key="0.5"
      >
        jalapeno
      </li>
    </ul>
  </div>
  <div
    className="order"
    key="0.5"
  >
    <h3>
      Sam
    </h3>
    <ul
      className="ingredient-list"
    >
      <li
        key="0.5"
      >
        steak
      </li>
      <li
        key="0.5"
      >
        pico de gallo
      </li>
      <li
        key="0.5"
      >
        lettuce
      </li>
      <li
        key="0.5"
      >
        carnitas
      </li>
      <li
        key="0.5"
      >
        queso fresco
      </li>
      <li
        key="0.5"
      >
        jalapeno
      </li>
    </ul>
  </div>
  <div
    className="order"
    key="0.5"
  >
    <h3>
      Alex
    </h3>
    <ul
      className="ingredient-list"
    >
      <li
        key="0.5"
      >
        sofritas
      </li>
      <li
        key="0.5"
      >
        beans
      </li>
      <li
        key="0.5"
      >
        sour cream
      </li>
      <li
        key="0.5"
      >
        carnitas
      </li>
      <li
        key="0.5"
      >
        queso fresco
      </li>
    </ul>
  </div>
</section>
`;
